xapian-core/include/xapian/cluster.h

```diff
@@ -28,14 +28,45 @@
 #endif
 
 #include <xapian/mset.h>
+#include <xapian/queryparser.h>
 #include <xapian/types.h>
 #include <xapian/visibility.h>
 
 #include <unordered_map>
+#include <unordered_set>
 #include <vector>
 
 namespace Xapian {
 
+/// Stopper subclass which checks for both stemmed and unstemmed stopwords
+class XAPIAN_VISIBILITY_DEFAULT StemStopper : public Xapian::Stopper {
+  public:
+    /// Stemming strategies
+    typedef enum { STEM_NONE, STEM_SOME, STEM_ALL_Z } stem_strategy;
```

Comment:
You should support the same members as for `TermGenerator::set_stemming_strategy()` and similar, because otherwise it's confusing. I believe that `STEM_ALL` could do `stop_words.insert(stemmer(term))` to behave consistently.